docs/McpClient-Feedback.md

```diff
@@ -0,0 +1,47 @@
+ModelContextProtocol.Protocol.Implementation name is too generic.  Lots of things in this namespace named poorly.  That might be OK if users don't have to deal with them, but it looks like they do.
+
+
+ModelClient.CompleteAsync - naming seems off.  Maybe "GetCompletion" 
+
+
+
+static ModelClient.CreateSamplingHandler - why is this factory method on ModelClient???  Maybe should be an extension method.
+
+
+
+ModelClient - Enumerate\* vs List\* - why have these redundant methods that return the same thing just one wrapping.  Can we just return IAE and have folks materialize that?
+
+
+
+ModelClient.EnumerateToolsAsync - why accept JsonSerializerOptions on this call, it doesn't feel like something tied to enumerating tools.  Would it make more sense to make this a property on the client?
+
+
+
+SetLoggingLevel(LogLevel, CancellationToken) vs SetLoggingLevel(LoggingLevel, CancellationToken) -- WHY?  Don't make these overloads, instead choose one then make conversions on the types.
+
+
+
+SubscribeToResourceAsync - how are the notifications delivered?  This method confused me as I don't understand what it's supposed to do.  Might just need to research more.
+
+
+TextContentBlock doesn't override ToString -- to get any of the data returned I need to get at protocol types, which feels wrong since those are all "raw" and not designed surface area.
```

Comment:
@jeffhandley -- here's an example of one of the "need to use Protocols" to do a basic thing.  I didn't do a deep scrub of everything, but this one was a **very basic** scenario

Reply:
To use these we have to downcast.  There is no DebuggerDisplay to help folks understand what they have in the IDE.  We could make these more closely match the pattern used in AIContent.   @ericstj to file issue - DebuggerDisplay and ToString.